src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -143,9 +163,17 @@ class MetadataBasedItemList extends React.Component<Props, State> {
         return headerData;
     }
 
-    handleMouseEnter = (rowIndex: number): void => this.setState({ hoveredRowIndex: rowIndex });
+    handleMouseEnter = (columnIndex: number, rowIndex: number): void =>
+        this.setState({
+            hoveredColumnIndex: columnIndex,
+            hoveredRowIndex: rowIndex,
+        });
 
-    handleMouseLeave = (): void => this.setState({ hoveredRowIndex: -1 });
+    handleMouseLeave = (): void =>
```

Comment:
Would it make sense to debounce this leave handler or is React's setState efficient enough that it will batch the changes to state based on leave and enter events so that no extra render happens?

Reply:
React event handlers are batched by default - https://stackoverflow.com/questions/48563650/does-react-keep-the-order-for-state-updates/48610973#48610973